modules/mtl/arrow-mtl-data/src/main/kotlin/arrow/mtl/typeclasses/MonadTrans.kt

```diff
@@ -0,0 +1,34 @@
+package arrow.mtl.typeclasses
+
+import arrow.Kind
+import arrow.Kind2
+import arrow.typeclasses.Monad
+
+/**
+ * MonadTrans is a typeclass that abstracts lifting arbitray monadic computations in another context.
+ */
+interface MonadTrans<T> {
+  /**
+   * transforms a given monad `Kind<F, A>` to `Kind2<T, F, A>`
+   *
+   * {: data-executable='true'}
+   *
+   * ```kotlin:ank
+   * import arrow.mtl.extensions.optiont.monadTrans.monadTrans
+   * import arrow.core.extensions.id.monad.monad
+   * import arrow.core.extensions.*
+   * import arrow.core.*
+   * import arrow.mtl.*
+   *
+   * fun main(args: Array<String>) {
+   *    // sampleStart
+   *    val result = OptionT.monadTrans().run {
+   *      Id.just("hello").lift(Id.monad())
+   *    }
+   *    // sampleEnd
+   *    println(result)
+   * }
+   * ```
+   */
+  fun <F, A> Kind<F, A>.lift(MF: Monad<F>): Kind2<T, F, A>
```

Comment:
```
fun <G, A> Kind<G, A>.lift(MF: Monad<G>): Kind2<F, G, A>
```

Reply:
If we're using `liftF` for concrete instances and we have `lift` defined in `Functor`, maybe it'd be better to use `liftF` here too.

Reply:
I think we should do this on case by case basis, because most of the time the current `lift` is equal to the one from `MonadTrans`. I think the only exception is `Kleisli` because it can lift any `Kind` not just monads (although it needs to be a monad so that Kleisli is a monad), Not sure what name to give in what case, but all other mtl types afaik need at least `Functor` to lift. (And given the fact that the only useful types to lift are monads we might as well use monad everywhere)

Reply:
We have `liftF` for Free, and similar others. `liftM`? `liftT`?

Reply:
I completely forgot that we have `liftF` for `Free` but that is in an entirely different context so not sure it matters. Honestly I'd stay with just `lift` here, if that won't work `liftT` for "liftTransformer" sounds best.

Reply:
i changed  it to liftT as this seemed to be the one both of you agreed on. if not, let me know its a quick change 😄